packages/store/node-server-sdk-redis/src/RedisClientState.ts

```diff
@@ -0,0 +1,128 @@
+import { LDLogger } from '@launchdarkly/node-server-sdk';
+import { Redis } from 'ioredis';
+import LDRedisOptions from './LDRedisOptions';
+
+const DEFAULT_PREFIX = 'launchdarkly';
+
+/**
+ * Class for managing the state of a redis connection.
+ *
+ * Used for the redis persistent store as well as the redis big segment store.
+ *
+ * @internal
+ */
+export default class RedisClientState {
+  private connected: boolean = false;
+
+  private attempt: number = 0;
+
+  private initialConnection: boolean = true;
+
+  private client: Redis;
+
+  private owned: boolean;
+
+  private base_prefix: string;
```

Comment:
Seems like these 3 can be readonly:

```suggestion
  private readonly client: Redis;

  private readonly owned: boolean;

  private readonly base_prefix: string;
```